osu.Game/Graphics/UserInterface/ShearedSliderBar.cs

```diff
@@ -193,8 +193,13 @@ protected override void UpdateAfterChildren()
         {
             base.UpdateAfterChildren();
 
-            LeftBox.Size = new Vector2(Math.Clamp(RangePadding + Nub.DrawPosition.X - Nub.DrawWidth / 2f + ShearedNub.CORNER_RADIUS - 0.5f, 0, Math.Max(0, DrawWidth)), 1);
-            RightBox.Size = new Vector2(Math.Clamp(DrawWidth - RangePadding - Nub.DrawPosition.X - Nub.DrawWidth / 2f + ShearedNub.CORNER_RADIUS - 0.5f, 0, Math.Max(0, DrawWidth)), 1);
+            float nubOriginOffset = Nub.OriginPosition.X - Nub.DrawWidth / 2f;
+            float nubPosition = Nub.DrawPosition.X - nubOriginOffset;
+            LeftBox.Size = new Vector2(
+                Math.Clamp(RangePadding + nubPosition - Nub.DrawWidth / 2f + ShearedNub.CORNER_RADIUS - 0.5f, 0, Math.Max(0, DrawWidth)), 1);
+            RightBox.Size = new Vector2(
+                Math.Clamp(DrawWidth - RangePadding - nubPosition - Nub.DrawWidth / 2f + ShearedNub.CORNER_RADIUS - 0.5f, 0,
```

Comment:
I'm not sure how you got to this code, but it reads very bad. For instance, the split out variables are being used in two places which are also subtracting `DrawWidth/2` a second time, I don't quite follow what is going on here anymore.